plugins/eos-blacklist/gs-plugin-eos-blacklist.c

```diff
@@ -207,19 +215,42 @@ app_get_flatpak_ref_name (GsApp *app)
 	return gs_app_get_metadata_item (app, "flatpak::RefName");
 }
 
+static gchar **
+get_default_locales (GsPlugin *plugin)
+{
+	g_auto(GStrv) flatpak_default_locales = NULL;
+	g_autoptr(GError) local_error = NULL;
+
+	GsPluginData *priv = gs_plugin_get_data (plugin);
+	if (priv->installation == NULL)
+		return NULL;
+
+	flatpak_default_locales = flatpak_installation_get_default_languages (priv->installation, &local_error);
+	if (local_error != NULL) {
+		g_warning ("No user locales could be retrieved! %s", local_error->message);
+		return NULL;
+	}
+
+	return g_steal_pointer (&flatpak_default_locales);
+}
+
 static gboolean
 gs_plugin_locale_is_compatible (GsPlugin *plugin,
 				const char *locale)
 {
 	g_auto(GStrv) locale_variants;
+	g_auto(GStrv) locale_options = NULL;
 	const char *plugin_locale = gs_plugin_get_locale (plugin);
-	int idx;
 
+        /* Check if a variant of a locale is compatible, e.g. 'pt' for 'pt_BR' */
 	locale_variants = g_get_locale_variants (plugin_locale);
-	for (idx = 0; locale_variants[idx] != NULL; idx++) {
-		if (g_strcmp0 (locale_variants[idx], locale) == 0)
-			return TRUE;
-	}
+	if (locale_variants != NULL && g_strv_contains ((const gchar * const *) locale_variants, locale))
+		return TRUE;
+
+        /* check if the app's locale is listed in the languages key for the ostree repo file */
+	locale_options = get_default_locales (plugin);
+	if (locale_options != NULL && g_strv_contains ((const gchar * const *) locale_options, locale))
```

Comment:
Perhaps a stupid question, but… do we not need to go through `g_get_locale_variants` for these locales too, in one or other direction? If one of the default locales is `en`, then I guess we'd expect `com.endlessm.times_of_india.en_IN` to match here?

Reply:
I think the direction should be: people with the default locale being `en_IN` and seeing the `com.endlessm.encyclopedia.en`, not the other way around, right? And I think the previous if would match that.

Reply:
Oh, of course, you mean the extra-language settings - since as of now this is strictly a configuration of the image personality, this will be pretty much targeted to a specific audience, so I don't think there's a need to derive the languages that way

Reply:
I just want to make sure this limitation is understood: as this code is currently implemented, only apps which **exactly** match an element of `xa.extra-languages` will be considered. We can and do use both one-part language suffixes (`.en`, `.pt`) and two-part locale suffixes (`.en_IN`, `.pt_BR`) on apps, so what we're implicitly saying here is that in the cases where we rely on `xa.extra-languages` to cause certain apps to be displayed, we will only use two-character language suffixes for the apps in question, and accept that apps with full locale suffixes will not be shown.

Reply:
To me, that sounds like a very subtle distinction which is going to get forgotten soon, and is going to cause someone half a day’s debugging of gnome-software to work out why an app isn’t being installed when it should.

Are there any downsides to supporting matching locale variants?

Reply:
Hmm no I don't think there are downsides to that - flatpak only allows setting one-part language suffixes (for both `xa.languages` and `xa.extra-languages`) so, the only way to make two-part language suffixes would be to whitelist them if the one-part counterpart is present on the list.